File: proytaller2/src/service/ClienteService.js
// src/service/service.js
import axios from "axios"

const API_URL = "http://localhost:9090/api/cliente"

const apiClient = axios.create({
  baseURL: API_URL,
  headers: {
    "Content-Type": "application/json"
  }
})
export default {
  //añadir cliente
  addClient(cliente){
    return apiClient.post("/anadir", cliente)
  },

  //eliminar cliente
  deleteCliente(idCliente){
    return apiClient.delete(`/eliminar-client/${idCliente}`)
  },

  //actualizar cliente
  updateClient(idCliente, cliente){
    return apiClient.put(`/update/${idCliente}`, cliente)
  },

  //buscar cliente
  buscarCliente(nombre){
    return apiClient.get(`/buscar-cliente?nombre=${nombre}`)
  },

  //Lista de todos los clientes
  listarClientes() {
    return apiClient.get("/listar-cliente");
  },

  //Obtener cliente por id
  obtenerClientePorId(idCliente){
    return apiClient.get(`/obtener-cliente/${idCliente}`)
  }

}

File: proyect/proyect/src/main/java/taller2backend/proyect/service/ClienteService.java
package taller2backend.proyect.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;

import taller2backend.proyect.dto.ClienteListDTO;
import taller2backend.proyect.entity.Cliente;

public interface ClienteService {
    @Autowired

    // buscar todos los datos
    public Cliente addClient(Cliente cliente);

    //eliminar cleente por id 
    public void deteleClient(Long idCliente);


    //actualizar datos del cliente
    public Cliente updateClient(Long idCliente, Cliente clienteNuevo);
    
//buscar cliente por nombre
    public List<ClienteListDTO> buscarClientPorName(String name);

    //Lista de todos los clientes
    public List<ClienteListDTO> listarClientes();
    
    //Obtener cliente por id
    public Cliente obtenerClientePorId(Long idCliente);
    // Lista de todos los clientes que tienen una ventas

}
